api/serializers.py

```diff
@@ -33,7 +36,28 @@ class Meta:
         fields = ('token', 'username', 'password')
 
 
+#NOTE:
+#There are currently a couple of issues here
+#1. The file needs a name, but not a title
 class VideoSerializer(serializers.ModelSerializer):
+
+    def create(self, validated_data):
+        data = self.context.get('data')
+        user = self.context.get('user')
+        #Data is prepended by "data:video/mp4;base64,", need to remove those 22 characters
+        #Encode converts from string to bytes
+        data_bytes = data[22:].encode('ascii')

```

Comment:
What if the data isn't mp4? Do we determine the prepended encoding and MIME type metadata? Have you had a look at [custom parsers](https://www.django-rest-framework.org/api-guide/parsers/#custom-parsers)?

Reply:
`react-media-recorder` leaves the data as a Blob, which is then converted into a base64 encoded string using FileReader.

Reply:
I think `FileUploadParser` would be of particular interest here

Reply:
Ah okay I see. Why send the data over as a blob over just straight binary? Seems like converting the blob to base64 instead of a typed array needlessly increases the data being sent by 150%

Reply:
Looks like there is a [file reader function](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsArrayBuffer) to read the data into an array

Reply:
I don't think the size difference justifies sending the data as raw binary instead of base64, but if the data is being saved as a blob, then the metadata should be saved along with it

Reply:
Also, it looks like the `ContentFile` object supports binary strings directly so there is no need to encode the blob as ascii.

Reply:
Let's revisit this discussion this week